src/faultlog/guard_with_eid_records.cpp

```diff
@@ -58,25 +60,136 @@ static int getGuardedTarget(struct pdbg_target* target, void* priv)
         }
-        count += 1;
+        auto physicalPath = openpower::guard::getPhysicalPath(elem.targetId);
+        if (!physicalPath.has_value())
+        {
+            lg2::error("Failed to get physical path for record {RECORD_ID}",
+                       "RECORD_ID", elem.recordId);
+            continue;
+        }
+
+        GuardedTarget guardedTarget(*physicalPath);
+        pdbg_target_traverse(nullptr, getGuardedTarget, &guardedTarget);
+        if (guardedTarget.target == nullptr)
+        {
+            lg2::error("Failed to find the pdbg target for the guarded "
+                       "target {RECORD_ID}",
+                       "RECORD_ID", elem.recordId);
+            continue;
+        }
+        ATTR_HWAS_STATE_Type hwasState;
+        if (DT_GET_PROP(ATTR_HWAS_STATE, guardedTarget.target, hwasState))
+        {
+            lg2::error("Failed to get HWAS state of the guarded "
+                       "target {RECORD_ID}",
+                       "RECORD_ID", elem.recordId);
+            continue;
+        }
+        bool dbusErrorObjFound = true;
+        uint32_t bmcLogId = 0;
+        try
+        {
+            auto method = bus.new_method_call(
+                "xyz.openbmc_project.Logging", "/xyz/openbmc_project/logging",
+                "org.open_power.Logging.PEL", "GetBMCLogIdFromPELId");
+
+            method.append(static_cast<uint32_t>(elem.elogId));
+            auto resp = bus.call(method);
+            resp.read(bmcLogId);
+        }
+        catch (const sdbusplus::exception::SdBusError& ex)
+        {
+            dbusErrorObjFound = false;
+            lg2::info(
+                "PEL might be deleted but guard entry is around {ELOG_ID)",
+                "ELOG_ID", elem.elogId);
+        }
+        uint32_t plid = 0;
+        if (dbusErrorObjFound)
+        {
+            Properties pelEntryProp;
+            std::string objPath = "/xyz/openbmc_project/logging/entry/" +
+                                  std::to_string(bmcLogId);
+            auto pelEntryMethod = bus.new_method_call(
+                "xyz.openbmc_project.Logging", objPath.c_str(),
+                "org.freedesktop.DBus.Properties", "GetAll");
+            pelEntryMethod.append("org.open_power.Logging.PEL.Entry");
+            auto pelEntryReply = bus.call(pelEntryMethod);
+            pelEntryReply.read(pelEntryProp);
+            for (const auto& [prop, propValue] : pelEntryProp)
+            {
+                if (prop == "PlatformLogID")
+                {
+                    auto plidPtr = std::get_if<uint32_t>(&propValue);
+                    if (plidPtr != nullptr)
+                    {
+                        plid = *plidPtr;
+                    }
+                }
+            }
+            // A single PEL could capture multiple guard records callouts so
+            // ignore a guard record if already cpatured as part of a PEL
+            if (std::find(liProcessedPels.begin(), liProcessedPels.end(),
```

Comment:
Can we push this outside the outer if and else block as we are anyway doing it for both cases.

Reply:
i cannot push outside as i need plid both for guard records with D-Bus error object found and not found

Reply:
yes makes sense pushed outside